lib/src/license_detection/tokenizer.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:string_scanner/string_scanner.dart';
+
+/// A token in this context is a sequence of characters
+/// scanned from the original text separated by a space
+/// or new line.
+@sealed
+class Token {
+  /// Normalized form of the text in [span].
+  final String value;
+
+  /// Zero based position of token in a list of [Token]s obtained by tokenizing license text.
+  ///
+  /// License when tokenized through [tokenize] method
+  /// yields a list of [Token]. [index] denotes the offset of the token
+  /// in this list.
+  final int index;
+
+  /// Zero based line number of token.
+  int get line => span.start.line;
+
+  /// Location details of the token in original text.
+  final SourceSpan span;
+
+  Token(this.value, this.index, this.span);
+
+  Token.fromSpan(this.span, this.index)
+      : value = _normalizeWord(span.text.toLowerCase());
+}
+
+/// Tokenizes license text and returns a list of [Token]s.
+///
+/// The tokens are scanned and normalized according to this procedure.
+/// 1. Tokens are words separated by spaces or new line. Ex - `hello new\nworld` -->[`hello`,`new`,`world`].
+/// 2. Any pure puntcuations texts are ignored. Ex - `// % ^&^&*^ hello` --> [`hello`]
+/// 3. Leading punctuations are ignored. Ex - `!hello $world&` --> [`hello`,`world&`]
+/// 4. New line tokens are stored to stored initially to identify list items,
+///    but later removed while cleaning tokens.
+/// 5. Any tokens starting with alphabet are further cleaned to remove any
+///    non-alphabet characters. Ex - `(hello wo$r1ld` --> [`hello`, `world`]
+/// 6. If a token starts with digit any characters other than `.`, `-` will be ignored.
```

Comment:
> If a token starts with digit any characters other than `.`, `-` will be ignored.

I don't understand what this means.

Reply:
`If a token starts with digit, any characters other than `.` `-` in the token value will be ignored.` It was supposed to sound like this but I think I made a mistake while typing.